### ISSUE TITLE
Change value of IHostingEnvironment.EnvironmentName (readonly property)

### ISSUE THREAD

**Issue by jsinh** [aspnet/Hosting#139]
I am trying to find out how can I can IHostingEnvironment.EnvironmentName value from my application.

I have gone through the code in `Microsoft.AspNet.Hosting` library and observed that class `ConfigureHostingEnvironment` and `HostingEnvironment` plays the main role in setting this up. If no value found it sets up `Development` as the default environment name.

`private const string DefaultEnvironmentName = "Development";`

In `ConfigureHostingEnvironment` class it tries to fetch `private const string EnvironmentKey = "ASPNET_ENV";` key to get user set environment name. As per my understanding, for this key it looks for `Microsoft.AspNet.Hosting.ini` file

`Program` class adds this 'Microsoft.AspNet.Hosting.ini' file to configuration instance and also environment variables to look for and command line to look for.

Based on above understanding I have tried all possible combination to change the value of `EnvironmentName` but didn't succeed. Can someone please help me figure out how to achieve this?

**Comment by davidfowl** [comment]
Setting it from your application is the wrong thing to do. You either set the environment variable or you set the value in the file.

**Comment by jsinh** [comment]
I agree with you and i observed that the value for `EnvironmentName` property is set to `Development` even before my constructor for `Startup.cs` starts. I tried doing both - adding value ASPNET_ENV = "FooBar" in `Microsoft.AspNet.Hosting.ini` file but didn't worked. Same I did by setting environment variable (System / User both) but that too didn't worked.

What is the exact key name / format to be used to make it work?
